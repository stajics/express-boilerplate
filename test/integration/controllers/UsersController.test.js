require('../../index');

describe(':editUser', () => {
  jestTest('[PUT /users/:uid] Should edit user 1', async (done) => {
    client.setToken(fixtures.users[0].token);
    const res = await client.put(`/users/${fixtures.users[0].uid}`, {
      firstName: 'changedFirstName',
    });
    expect(res.body).toMatchSnapshot();
    done();
  });

  jestTest('[PUT /users/:uid] Should error (not authorized, editing some other user)', async (done) => {
    client.setToken(fixtures.users[0].token);
    const res = await client.put(`/users/${fixtures.users[1].uid}`, {
      firstName: 'changedFirstName',
    });
    expect(res.body).toMatchSnapshot();
    done();
  });
});
