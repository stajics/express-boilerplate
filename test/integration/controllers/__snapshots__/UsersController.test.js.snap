// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`:editUser [PUT /users/:uid] Should edit user 1 1`] = `
Object {
  "changedAttributes": Object {
    "firstName": "changedFirstName",
  },
  "text": "Edited user.",
}
`;

exports[`:editUser [PUT /users/:uid] Should error (not authorized, editing some other user) 1`] = `
Object {
  "error": Object {
    "code": 4,
    "message": "Unauthorized.",
    "status": 401,
  },
}
`;
